src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -1189,10 +1189,19 @@ bool IFloatingPoint<decimal>.TryWriteSignificandLittleEndian(Span<byte> destinat
         {
             if (destination.Length >= (sizeof(ulong) + sizeof(uint)))
             {
+                var lo64 = _lo64;
+                var hi32 = _hi32;
+
+                if (!BitConverter.IsLittleEndian)
+                {
+                    lo64 = BinaryPrimitives.ReverseEndianness(lo64);
+                    hi32 = BinaryPrimitives.ReverseEndianness(hi32);
+                }
+
                 ref byte address = ref MemoryMarshal.GetReference(destination);
 
-                Unsafe.WriteUnaligned(ref address, _lo64);
-                Unsafe.WriteUnaligned(ref Unsafe.AddByteOffset(ref address, sizeof(ulong)), _hi32);
+                Unsafe.WriteUnaligned(ref address, lo64);
+                Unsafe.WriteUnaligned(ref Unsafe.AddByteOffset(ref address, sizeof(ulong)), hi32);
```

Comment:
Not sure if this is fully correct either.  As I understand it, the "_lo64" field in the data structure is a composite of the logically separate "lo" and "mid" components.  On a little-endian platform, the current code outputs the three fields in the order "lo", "mid", and "hi" - or equivalently, it outputs the entire significand viewed as a single little-endian 96-bit integer.

This new code for big-endian platforms now does neither, but outputs the components in the order "mid", "lo", "hi", each considered as a 32-bit big-endian value.   I'm not sure this is what is expected?

Reply:
Say we have a 96-bit integer `0xC0B0_A090_8070_6050_4030_2010`

This will be split into:
* `hi32`: `0xC0B0_A090`
* `lo64`: `0x8070_6050_4030_2010`

On a little-endian platform, this will be:
* `hi32`: `0x90`, `0xA0`, 0xB0`, `0xC0
* `lo64`: `0x10`, `0x20`, `0x30`, `0x40`, `0x50`, `0x60`, `0x70`, `0x80`

This exactly matches the output we want.

On a big-endian platform, this will be inverse:
* `hi32`: `0xC0`, `0xB0`, `0xA0`, `0x90`
* `lo64`: `0x80`, `0x70`, `0x60`, `0x50`, `0x40`, `0x30`, `0x20`, `0x10`

If we reverse each part and still write them as `lo64`, `hi64`, then the total set will still be ordered correctly.

-----------------

That is to say, `lo64` isn't treated as "separate" `low32`, `mid32` fields. It's treated as a contiguous 64-bits where the upper 32-bits are always `mid32`.

Reply:
Ah, right, sorry for the confusion.  I was misinterpreting how the lo64 field is read in.   The patch should indeed be correct.  Thanks!

Reply:
No worries, it's definitely a place that's easy to get mixed up (particularly since there could've been several different implementations here).